pinot-spi/src/main/java/org/apache/pinot/spi/data/Schema.java

```diff
@@ -299,6 +299,24 @@ public DateTimeFieldSpec getDateTimeSpec(String dateTimeName) {
     return null;
   }
 
+  /**
+   * Fetches the DateTimeFieldSpec for the given time column name.
+   * If the columnName is a DATE_TIME column, returns the DateTimeFieldSpec
+   * If the columnName is a TIME column, converts to DateTimeFieldSpec before returning
+   */
+  public DateTimeFieldSpec getSpecForTimeColumn(String timeColumnName) {
+    FieldSpec fieldSpec = _fieldSpecMap.get(timeColumnName);
+    if (fieldSpec != null) {
+      if (fieldSpec.getFieldType() == FieldType.DATE_TIME) {
+        return (DateTimeFieldSpec) fieldSpec;
+      }
+      if (fieldSpec.getFieldType() == FieldType.TIME) {
+        return convertToDateTimeFieldSpec((TimeFieldSpec) fieldSpec);
+      }
+    }
+    return null;
```

Comment:
Can we throw IllegalStateException here?

Reply:
I'm trying to maintain the behavior exhibited by the old method that this new one will replace. Marked the method Nullable as an additional measure.